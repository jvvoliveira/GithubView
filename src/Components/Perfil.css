.title{
    text-align: center
}
.box{
    display: inline-block;
    margin: 10px;
}