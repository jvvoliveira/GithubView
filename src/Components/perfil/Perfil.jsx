import React, { useContext } from "react";
import styles from "./Perfil.module.scss";
import { loadingContext } from "../../App";

const Perfil = () => {
  const { data, status } = useContext(loadingContext).usuario;

  if (data !== undefined) {
    if (status === "OK") {
      return (
        <div className={styles.container}>
          <h1 data-testid='name'>{data.name ? data.name : data.login}</h1>
          <div className={styles.wrapper}>
            <div className={styles.wrapper_img}>
              <a href={data.html_url} target="_blank">
                <img src={data.avatar_url} />
              </a>
            </div>
            <div className={styles.wrapper_info}>
              <p data-testid='bio'>{data.bio}</p>
              <p>{data.location}</p>
              <p data-testid='followers'>Seguidores: {data.followers}</p>
              <p>Repositórios públicos: {data.public_repos}</p>
              <p>Criado em: {data.created_at}</p>
            </div>
          </div>
        </div>
      );
    } else {
      return <h2 data-testid="messageNotFound">Usuário não encontrado</h2>;
    }
  } else {
    return <h2 data-testid="messageInitial">Pesquise por algum usuário GitHub</h2>;
  }
};

export default Perfil;