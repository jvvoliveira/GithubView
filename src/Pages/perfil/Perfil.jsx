import React, { useContext } from "react";
import styles from "./Perfil.module.scss";
import { loadingContext } from "../../App";
import EmptyContent from "../../Components/emptyContent/EmptyContent";

const Perfil = () => {
  const { data, status } = useContext(loadingContext).usuario;

  if (data === undefined)
    return <EmptyContent user dataTestId="messageInitial" message="Pesquise por algum usuário GitHub" />

  if (status !== "OK")
    return <EmptyContent user dataTestId="messageNotFound" message="Usuário não encontrado" />

  return (
    <div className={styles.container}>
      <h1 data-testid="nome">{data.name ? data.name : data.login}</h1>
      <div className={styles.wrapper}>
        <div data-testid="imagemPerfil" className={styles.wrapper_img}>
          <a href={data.html_url} target="_blank">
            <img data-testid="foto" src={data.avatar_url} />
          </a>
        </div>
        <div
          className={styles.wrapper_info}
        >
          <p data-testid="bio">{data.bio}</p>
          <p data-testid="local">{data.location}</p>
          <p data-testid="seguidores">Seguidores: {data.followers}</p>
          <p data-testid="reposPublicos">
            Repositórios públicos: {data.public_repos}
          </p>
          <p data-testid="dataCriacao">Criado em: {data.created_at}</p>
        </div>
      </div>
    </div>
  )
};

export default Perfil;
