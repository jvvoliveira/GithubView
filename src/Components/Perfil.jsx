import React, { useContext } from 'react'
import './Perfil.css'
import { loadingContext } from '../App'

export default () => {
    const {usuario} = useContext(loadingContext);

    if (usuario !== undefined) {
        return (
            <>
                <h1 className='title'>{usuario.name ? usuario.name : usuario.login}</h1>
                <div className='box'>
                    <img src={usuario.avatar_url}></img>
                </div>
                <div className='box'>
                    <p>{usuario.bio}</p>
                    <p>{usuario.location}</p>
                    <p>Seguidores: {usuario.followers}</p>
                    <p>Repositórios públicos: {usuario.public_repos}</p>
                    <h4><a href={usuario.html_url} target="_blank">GitHub</a></h4>
                    <p>Criado em: {usuario.created_at}</p>
                </div>
            </>
        )
    } else {
        return (
            <h2>Sem usuário</h2>
        )
    }
}