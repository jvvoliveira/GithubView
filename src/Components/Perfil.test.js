import React from 'react'
import ReactDOM from 'react-dom';
import Perfil, { pesquisar } from './Perfil'
import renderer from 'react-test-renderer';
import {
    render,
    fireEvent,
    cleanup,
    waitForElement,
    createEvent
} from '@testing-library/react';

// describe('teste do perfil', () => {
//     it('should renderer perfil', () => {
//         const div = document.createElement('div');
//         ReactDOM.render(<Perfil usuario={ usuario = {url: 'url'} }/>, div);
//         ReactDOM.unmountComponentAtNode(div);
//     }),
//     it('snapshot view perfil', () => {
//         const tree = renderer.create(<Perfil/>).toJSON();
//         expect(tree).toMatchSnapshot();
//     }),
//     it('should ', async () => {
//         const setUsuario = jest.fn()
//         const json = () => {
//             return new Promise((resolve) => {
//                 resolve('mock perfil');
//             })
//         }
//         const request = () => {
//             return new Promise((resolve) => {
//                 resolve({ json, status: 200 });
//             })
//         }

//         await pesquisar('url', setUsuario, request)
//         expect(setUsuario).toBeCalledWith('mock perfil')
//     })
// })