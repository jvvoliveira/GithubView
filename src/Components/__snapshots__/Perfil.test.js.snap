// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste do perfil snapshot view perfil 1`] = `
<h2>
  Sem usuário ou
  <br />
  sem repositórios públicos
</h2>
`;
